core/src/main/java/com/novoda/noplayer/exoplayer/MediaSourceFactory.java

```diff
@@ -36,6 +38,10 @@ MediaSource create(ContentType contentType,
                         handler,
                         eventListener
                 );
+            case DASH:
+                return new DashMediaSource(uri, mediaDataSourceFactory,
+                                           new DefaultDashChunkSource.Factory(mediaDataSourceFactory), handler, mediaSourceEventListener
```

Comment:
this new can go in a local variable inside the `case` so that your return is smaller